internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/WriteComputeWithActiveActiveReplicationTest.java

```diff
@@ -0,0 +1,1184 @@
+    val1.put(INT_ARRAY_FIELD, Collections.emptyList());
+    val1.put(MAP_FIELD, mapFieldValue);
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(0)), storeName, key1, val1);
+    verifyFLMRecord(storeName, dc1RouterUrl, key1, regularFieldValue, Collections.emptyList(), mapFieldValue);
+    verifyFLMRecord(storeName, dc0RouterUrl, key1, regularFieldValue, Collections.emptyList(), mapFieldValue);
+
+    // PartialPut after DELETE: should be able to do PartialPut on the previously deleted Key-Value
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(0)), storeName, key1, null);
+    verifyFLMRecord(storeName, dc1RouterUrl, key1, null, null, null);
+    verifyFLMRecord(storeName, dc0RouterUrl, key1, null, null, null);
+    ub = new UpdateBuilderImpl(wcSchemaV1);
+    mapFieldValue = new HashMap<>();
+    mapFieldValue.put("ten", 10);
+    mapFieldValue.put("twenty", 20);
+    ub.setNewFieldValue(MAP_FIELD, mapFieldValue);
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(1)), storeName, key1, ub.build());
+    verifyFLMRecord(storeName, dc1RouterUrl, key1, regularFieldDefault, intArrayFieldDefault, mapFieldValue);
+    verifyFLMRecord(storeName, dc0RouterUrl, key1, regularFieldDefault, intArrayFieldDefault, mapFieldValue);
+
+    // AddToMap after DELETE: should be able to add a key to a map field of the previously deleted Key-Value
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(0)), storeName, key1, null);
+    verifyFLMRecord(storeName, dc1RouterUrl, key1, null, null, null);
+    verifyFLMRecord(storeName, dc0RouterUrl, key1, null, null, null);
+    ub = new UpdateBuilderImpl(wcSchemaV1);
+    mapFieldValue = new HashMap<>();
+    mapFieldValue.put("thirty", 30);
+    ub.setEntriesToAddToMapField(MAP_FIELD, mapFieldValue);
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(1)), storeName, key1, ub.build());
+    verifyFLMRecord(storeName, dc1RouterUrl, key1, regularFieldDefault, intArrayFieldDefault, mapFieldValue);
+    verifyFLMRecord(storeName, dc0RouterUrl, key1, regularFieldDefault, intArrayFieldDefault, mapFieldValue);
+
+    // RemoveFromMap after DELETE: should get a record with default values
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(0)), storeName, key1, null);
+    verifyFLMRecord(storeName, dc1RouterUrl, key1, null, null, null);
+    verifyFLMRecord(storeName, dc0RouterUrl, key1, null, null, null);
+    ub = new UpdateBuilderImpl(wcSchemaV1);
+    ub.setKeysToRemoveFromMapField(MAP_FIELD, Collections.singletonList("fortyTwo"));
+    sendStreamingRecord(systemProducerMap.get(childDatacenters.get(1)), storeName, key1, ub.build());
+    verifyFLMRecord(storeName, dc1RouterUrl, key1, regularFieldDefault, intArrayFieldDefault, mapFieldDefault);
+    verifyFLMRecord(storeName, dc0RouterUrl, key1, regularFieldDefault, intArrayFieldDefault, mapFieldDefault);
+
+    /* PART B: Tests with logical timestamp
```

Comment:
btw do you think if this part A/ part B can be merged into one code and use test parameter to control it. It is fine if you think that's too much changes in single PR, we can revisit that in the near future.